var/spack/repos/builtin/packages/py-scipy/package.py

```diff
@@ -90,58 +83,65 @@ class PyScipy(PythonPackage):
-    )
-    depends_on("py-numpy@1.16.5:+blas+lapack", when="@1.6:1.6.1", type=("build", "link", "run"))
-    depends_on("py-numpy@1.14.5:+blas+lapack", when="@1.5.0:1.5", type=("build", "link", "run"))
-    depends_on("py-numpy@1.13.3:+blas+lapack", when="@1.3:1.4", type=("build", "link", "run"))
-    depends_on("py-numpy@1.8.2:+blas+lapack", when="@:1.2", type=("build", "link", "run"))
+    depends_on("py-numpy@1.21.6:1.27", when="@1.11:", type=("build", "link", "run"))
+    depends_on("py-numpy@1.19.5:1.26", when="@1.10", type=("build", "link", "run"))
+    depends_on("py-numpy@1.18.5:1.25", when="@1.9", type=("build", "link", "run"))
+    depends_on("py-numpy@1.17.3:1.24", when="@1.8", type=("build", "link", "run"))
+    depends_on("py-numpy@1.16.5:1.22", when="@1.6:1.7", type=("build", "link", "run"))
+    depends_on("py-numpy@1.14.5:1.21", when="@1.5", type=("build", "link", "run"))
+    depends_on("py-numpy@1.13.3:1.21", when="@1.3:1.4", type=("build", "link", "run"))
+    depends_on("py-numpy@1.8.2:1.20", when="@:1.2", type=("build", "link", "run"))
     depends_on("py-pytest", type="test")
 
-    # NOTE: scipy should use the same BLAS/LAPACK as numpy.
-    # For scipy 1.8 and older, this is achieved by calling the set_blas_lapack()
-    # and setup_build_environment() from numpy in the scipy spec.
-    depends_on("blas")
-    depends_on("lapack")
+    # Required to use --config-settings
+    depends_on("py-pip@23.1:", when="@1.9:", type="build")
 
     # https://docs.scipy.org/doc/scipy/dev/toolchain.html#other-libraries
     depends_on("lapack@3.7.1:", when="@1.9:")
     depends_on("lapack@3.4.1:", when="@1.2:")
+    depends_on("lapack")
+    depends_on("blas")
 
+    # meson.build
     # https://docs.scipy.org/doc/scipy/dev/toolchain.html#compilers
-    conflicts("%gcc@:7", when="@1.10:")
-    conflicts("%gcc@:4.7", when="@:1.9")
-    conflicts("%apple-clang@:9", when="@1.10:")
-    conflicts("%msvc@:19.19", when="@1.10:")
+    conflicts("%gcc@:7", when="@1.10:", msg="SciPy requires GCC >= 8.0")
+    conflicts("%gcc@:4.7", when="@:1.9", msg="SciPy requires GCC >= 4.8")
+    conflicts(
+        "%msvc@:19.19",
+        when="@1.10:",
+        msg="SciPy requires at least vc142 (default with Visual Studio 2019) "
+        "when building with MSVC",
+    )
 
-    # https://github.com/scipy/scipy/pull/11324
-    conflicts("@1.4.0:1.4.1", when="target=ppc64le:")
+    # https://github.com/scipy/scipy/issues/19352
+    conflicts("^py-cython@3.0.3")
 
     # https://github.com/mesonbuild/meson/pull/10909#issuecomment-1282241479
     # Intel OneAPI ifx claims to support -fvisibility, but this does not work.
     # Meson adds this flag for all Python extensions which include Fortran code.
     conflicts("%oneapi@:2023.0", when="@1.9:")
 
+    # error: expected unqualified-id (exact compiler versions unknown)
+    conflicts("%apple-clang@15:", when="@:1.9")
```

Comment:
@rgommers any idea what version of Apple Clang might have broken this?

Reply:
14 or 15 I think. We were on Clang 11/12 for a long time, and that always works. 13 got skipped somehow I believe, I can't remember building with it anywhere.